src/features/favorites/components/FavoritesResults.tsx

```diff
@@ -20,32 +19,71 @@ import { ColorsEnum, getSpacing, Spacer, TAB_BAR_COMP_HEIGHT } from 'ui/theme'
 
 const keyExtractor = (item: FavoriteResponse) => item.id.toString()
 
-export const FavoritesResults: React.FC = React.memo(() => {
+const useFetchFavorites = (data: PaginatedFavoritesResponse | undefined) => {
+  return (page: number, limit = 10): Array<FavoriteResponse> => {
+    if (!data) {
+      return []
+    }
+    return data.favorites.slice((page - 1) * limit, page * limit)
+  }
+}
+
+const useFavoritesResult = (limit = 10) => {
+  const [page, setPage] = useState(1)
+  // default this to true to kick the initial effect hook to
+  // fetch the first page
+  const [shouldFetch, setShouldFetch] = useState(true)
+  const [favorites, setFavorites] = useState<Array<FavoriteResponse>>([])
+  const { data, isFetching } = useFavorites()
+  const fetchFavorites = useFetchFavorites(data)
+  const fetchMore = useCallback(() => setShouldFetch(true), [])
+  useEffect(
+    () => {
+      // prevent fetching for other state changes
+      if (!shouldFetch) {
+        return
+      }
+
+      const fetch = () => {
+        const newFavorites = fetchFavorites(page, limit)
+
+        // set the should fetch call to false to prevent fetching
+        // on page number update
+        setShouldFetch(false)
+        setFavorites((oldFavorites) => [...oldFavorites, ...newFavorites])
+
+        //increment page for the next call
+        setPage(page + 1)
+      }
+
+      fetch()
+    },
+    // prevent fetching for other state changes
+    [page, shouldFetch]
+  )
```

Comment:
Ce block est un peu trop aéré du coup :D : 
```suggestion
  useEffect(() => {
      // prevent fetching for other state changes
      if (!shouldFetch) {
        return
      }

      function fetch() {
        const newFavorites = fetchFavorites(page, limit)
        // set the should fetch call to false to prevent fetching on page number update
        setShouldFetch(false)
        setFavorites((oldFavorites) => [...oldFavorites, ...newFavorites])
        // increment page for the next call
        setPage(page + 1)
      }
      fetch()
    },
    [page, shouldFetch]
  )
```

Reply:
OK